sydent/http/servlets/msisdnservlet.py

```diff
@@ -130,15 +134,27 @@ def render_GET(self, request):
 
     @jsonwrap
     def render_POST(self, request):
-        send_cors(request)
-
         authIfV2(self.sydent, request)
 
-        args = get_args(request, ('token', 'sid', 'client_secret'))
+        return self.do_validate_request(request)
 
-        return self.do_validate_request(args)
+    def do_validate_request(self, request):
+        """
+        Extracts information about a validation session from the request and
+        attempts to validate that session.
+
+        :param request: The request to extract information about the session from.
+        :type request: twisted.web.server.Request
+
+        :return: A dict with a "success" key which value indicates whether the
+            validation succeeded. If the validation failed, this dict also includes
+            a "errcode" and a "error" keys which include information about the failure.
+        :rtype: dict[str, bool or str]
+        """
+        send_cors(request)
```

Comment:
We probably want to keep this at the top, in case `authIfV2` returns/raises?

Reply:
I did that to reproduce how it's done in `emailservlet`, should I change both then?

Reply:
Yeah, please. Ideally we'd do this in the HTTP layer before we even call the handlers, but let's just move it back for now